specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2024-09-01-preview/appconfiguration.json

```
@@ -0,0 +1,3168 @@
+          {
+            "name": "regenerateKeyParameters",
+            "in": "body",
+            "description": "The parameters for regenerating an access key.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RegenerateKeyParameters"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request was successful; the request was well-formed and received properly.",
+            "schema": {
+              "$ref": "#/definitions/ApiKey"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ConfigurationStores_RegenerateKey": {
+            "$ref": "./examples/ConfigurationStoresRegenerateKey.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AppConfiguration/configurationStores/{configStoreName}/generateSasToken": {
+      "post": {
+        "tags": [
+          "ConfigurationStores"
+        ],
+        "description": "Generates the SAS token for scoped, read-only access of the specified configuration store.",
+        "operationId": "ConfigurationStores_GenerateSasToken",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ConfigStoreNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "generateSasTokenParameters",
+            "in": "body",
+            "description": "The object containing information for the SAS token generation request.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/GenerateSasTokenParameters"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request was successful; the request was well-formed and received properly.",
+            "schema": {
+              "$ref": "#/definitions/SasTokenGenerateResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ConfigurationStores_ListKeys": {
+            "$ref": "./examples/ConfigurationStoresGenerateSasToken.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AppConfiguration/configurationStores/{configStoreName}/resetSas": {
```

The term SAS is a bit incomplete. SAS token is more complete. I'd suggest the following API update.

```suggestion
    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AppConfiguration/configurationStores/{configStoreName}/resetSasTokens": {
```

cc @avanigupta